code/mob/living/critter/small_animal.dm

```diff
@@ -447,22 +459,113 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 		if (prob(10))
 			src.audible_message("[src] purrs!",\
 			"You purr!")
+		if (src.ai.enabled && ispug(user) && prob(10))
+			src.recent_pug_pet = TRUE
+			SPAWN(15 SECONDS)
+				src?.recent_pug_pet = FALSE
+			src.ai.priority_tasks += src.ai.get_instance(/datum/aiTask/sequence/goalbased/critter/attack, list(src, src.ai.default_task))
+			src.ai.interrupt()
+			src.visible_message("<span class='notice'>[src] recoils and hisses at [user]'s attempt to pet them, then goes for the jugular!</span>")
+			playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+
+	Crossed(atom/movable/M as mob)
+		..()
+		if (!isalive(src))
+			return
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(prob(10) && src.is_annoying)
```

Comment:
all cats are annoying, also this could be much higher given it's not a hard stun anymore

Reply:
Im not sure about that, since if all cats are capable of tripping on Cross() you run the risk of player controlled cats purposefully running into people's legs. So either i keep it the way it is, or i make all cats trip you but only if the ai is enabled.

Reply:
ooh right, that makes sense